test/main.test.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-env jest */
+
+import { createHiddenElement, removeElement, getHtmlUsingHiddenElement } from '../src/html-from-paste-event'
+
+it('creates a hidden contenteditable div', () => {
+  const element = createHiddenElement()
+
+  expect(element).toMatchSnapshot(' <div contenteditable="true" style="position: absolute; top:0; left: 0; opacity: 0;" />')
+
+  element.parentNode.removeChild(element)
+})
+
+it('removes a hidden contenteditable div', () => {
+  const element = createHiddenElement()
+  removeElement(element)
+
+  expect(document.body.childNodes.length).toEqual(0)
+})
+
+it('gets html using a hidden contenteditable div', () => {
+  document.execCommand = function () {}
+  const element = document.createElement('div')
+
+  const execCommandSpy = jest.spyOn(document, 'execCommand').mockImplementation(() => {
+    element.insertAdjacentHTML('afterbegin', '<h1>Title</h1><p>Paragraph</p>')
+  })
+
+  const html = getHtmlUsingHiddenElement(element)
+
+  expect(execCommandSpy).toHaveBeenCalled()
+  expect(html).toEqual('<h1>Title</h1><p>Paragraph</p>')
+})
```

Comment:
Thanks for putting these tests together 👍  we seem to be missing one for the exported method.

I've also had a thought about how we could actually only export `htmlFromPasteEvent` and keep the other functions private and still test it.

So if we mock `window.clipboardData` to ensure user goes down that code path. Then that is executed. We then hit a problem once we get to `document.execCommand` like we discussed yesterday. 

I think we could get around this problem though by if we mock `document.execCommand` with a jest function implementation that finds the last element on the dom (our hidden one) and sets the innerHTML to the content we expect pasted

What do you think? It'd resolve the need for our semi private functions

Reply:
Am I wrong thinking that the functionality of `htmlFromPasteEvent` is captured by #7?